src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java

```diff
@@ -1229,7 +1229,21 @@ void encodeQueryAsKvp(
         Filter transformedFilter = filter;
         if (declaredCRS != null) {
             transformedFilter =
-                    WFSReprojectionUtil.normalizeFilterCRS(filter, source.getSchema(), declaredCRS);
+                    WFSReprojectionUtil.normalizeFilterCRS(
+                            filter, source.getSchema(), declaredCRS, declaredCRS);
+        } else {
+            // let's try to sue the feature type info CRS
```

Comment:
Don't think we want to "sue" anyone here? :-p
Maybe "use"

Reply:
I'd also add a comment that this happens for example with complex features

Reply:
:P thx.